File: CS/ViewModels/DocumentViewModel.cs
using System.Windows;
using DXRibbonNotepad.Model;

namespace DXRibbonNotepad.ViewModels {
    public class DocumentViewModel : ViewModelBase {
        public Document Document {
            get { return document; }
            private set { document = value; }
        }
        public string Text {
            get { return Document.Text; }
            set {
                if(Text == value) return;
                Document.Text = value;
                OnPropertyChanged("Text");
            }
        }
        public string DocumentName {
            get { return Document.DocumentName; }
            set {
                if(DocumentName == value) return;
                Document.DocumentName = value;
                OnPropertyChanged("DocumentName");
            }
        }
        public string FontFamily {
            get { return Document.FontFamily; }
            set {
                if(Document.FontFamily == value) return;
                Document.FontFamily = value;
                OnPropertyChanged("FontFamily");
            }
        }
        public bool IsBold {
            get { return Document.IsBold; }
            set {
                if(Document.IsBold == value) return;
                Document.IsBold = value;
                OnPropertyChanged("IsBold");
            }
        }
        public bool IsItalic {
            get { return Document.IsItalic; }
            set {
                if(Document.IsItalic == value) return;
                Document.IsItalic = value;
                OnPropertyChanged("IsItalic");
            }
        }
        public bool IsUnderline {
            get { return Document.IsUnderline; }
            set {
                if(Document.IsUnderline == value) return;
                Document.IsUnderline = value;
                OnPropertyChanged("IsUnderline");
            }
        }
        public TextAlignment TextAlignment {
            get { return Document.TextAlignment; }
            set {
                if(Document.TextAlignment == value) return;
                Document.TextAlignment = value;
                OnPropertyChanged("TextAlignment");
            }
        }
        public double FontSize {
            get { return Document.FontSize; }
            set {
                if(Document.FontSize == value) return;
                Document.FontSize = value;
                OnPropertyChanged("FontSize");
            }
        }

        public DocumentViewModel(Document document) {
            this.document = document;
        }
        Document document;
    }
}

File: VB/ViewModels/DocumentViewModel.vb
Imports System.Windows
Imports DXRibbonNotepad.Model

Namespace DXRibbonNotepad.ViewModels

    Public Class DocumentViewModel
        Inherits ViewModelBase

        Public Property Document As Document
            Get
                Return documentField
            End Get

            Private Set(ByVal value As Document)
                documentField = value
            End Set
        End Property

        Public Property Text As String
            Get
                Return Document.Text
            End Get

            Set(ByVal value As String)
                If Equals(Text, value) Then Return
                Document.Text = value
                OnPropertyChanged("Text")
            End Set
        End Property

        Public Property DocumentName As String
            Get
                Return Document.DocumentName
            End Get

            Set(ByVal value As String)
                If Equals(DocumentName, value) Then Return
                Document.DocumentName = value
                OnPropertyChanged("DocumentName")
            End Set
        End Property

        Public Property FontFamily As String
            Get
                Return Document.FontFamily
            End Get

            Set(ByVal value As String)
                If Equals(Document.FontFamily, value) Then Return
                Document.FontFamily = value
                OnPropertyChanged("FontFamily")
            End Set
        End Property

        Public Property IsBold As Boolean
            Get
                Return Document.IsBold
            End Get

            Set(ByVal value As Boolean)
                If Document.IsBold = value Then Return
                Document.IsBold = value
                OnPropertyChanged("IsBold")
            End Set
        End Property

        Public Property IsItalic As Boolean
            Get
                Return Document.IsItalic
            End Get

            Set(ByVal value As Boolean)
                If Document.IsItalic = value Then Return
                Document.IsItalic = value
                OnPropertyChanged("IsItalic")
            End Set
        End Property

        Public Property IsUnderline As Boolean
            Get
                Return Document.IsUnderline
            End Get

            Set(ByVal value As Boolean)
                If Document.IsUnderline = value Then Return
                Document.IsUnderline = value
                OnPropertyChanged("IsUnderline")
            End Set
        End Property

        Public Property TextAlignment As TextAlignment
            Get
                Return Document.TextAlignment
            End Get

            Set(ByVal value As TextAlignment)
                If Document.TextAlignment = value Then Return
                Document.TextAlignment = value
                OnPropertyChanged("TextAlignment")
            End Set
        End Property

        Public Property FontSize As Double
            Get
                Return Document.FontSize
            End Get

            Set(ByVal value As Double)
                If Document.FontSize = value Then Return
                Document.FontSize = value
                OnPropertyChanged("FontSize")
            End Set
        End Property

        Public Sub New(ByVal document As Document)
            documentField = document
        End Sub

        Private documentField As Document
    End Class
End Namespace
